core/fortran-example/tutorial/02_tuple/tuple_example.f90

```diff
@@ -0,0 +1,59 @@
+SUBROUTINE tupleExample (part) BIND(C,name='tupleExample')
+  USE, INTRINSIC :: ISO_C_BINDING
+  implicit none
+  integer i,j
+  
+  !The Fortran derived type has the same memory layout as the C struct defined by
+  ! struct local_data_struct_t {     
+  !   double d0[3][3];     
+  !   double d1[4];     
+  !   int    d2; 
+  ! };
+  
+  type, BIND(C) :: ptl_type      
+     real (C_DOUBLE) :: d0(3,3) 
+     real (C_FLOAT ) :: d1(4) 
+     integer (C_INT) :: d2
+  end type ptl_type
+
+  type(ptl_type) :: part
```

Comment:
so here why can't we create a generic c pointer with an `allocate` function, do this cast, do the work, and then delete?

Reply:
May be possible, but I am not sure exactly how to do that.

Reply:
I think it may boil down to the fact that we know how to cast Cabana data structure to C struct, but not the other way around. @rfbird @ascheinb, any insight on this?

Reply:
I'm not actually sure. Can you expand what you guys mean exactly / are trying to achieve? 

@guangyechen if you can show me the cast you want, I can think about ways of trying to do it

Reply:
Per our last conference call, we will figure out the answer in another pull request.